### DISCUSSION TITLE
expose / show variables value on hover rather than Object:ID xxxxx while debug

### DISCUSSION THREAD

**Discussion by jzq740176597** [godotengine/godot-proposals#6631] [Editor]
### Describe the project you are working on

3D

### Describe the problem or limitation you are having in your project

while Debug, Most varriables I want to check value  unfortunatelly show "[object: Id xxxx]" on hover or stack-variable dock;

sometimes I can check vars by click sometimes not. CAN'T show anything , it's very unfriendly for debug;

I just want to check the variable **even it's a script-type on hover. I**t's easy I think,

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

see above

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

*

### If this enhancement will not be used often, can it be worked around with a few lines of script?

*

### Is there a reason why this should be core and not an add-on in the asset library?

core

**Comment by YuriSizov** [comment]
If you're seeing "Object:ID" then you're trying to inspect a complex object. There is nothing to display on hover, as it typically does not have a string representation. You should click on it to expand the detailed view.

**Comment by theraot** [comment]
I wish I could hover with the mouse, and the popup aside from the text also had a button to show it on the inspector or debugger panel accordingly. *Only while paused on the debugger, of course.*

Edit to clarify: I can find whatever it is in the debug panel, and if it is an object there will be a button on the debug panel to bring it to the inspector. While debugging I might have to do that too often, so I want a shorthand from hover to the inspector. However, for non-objects showing it in the debugger panel is a fine alternative.

**Comment by hsandt** [comment]
Related: my own suggestion "Debug: Code editor: command to debug content of Object variable under cursor/caret in remote Inspector" https://github.com/godotengine/godot-proposals/issues/9102
